src/hooks/usePendingSwapData.ts

```diff
@@ -517,11 +521,8 @@ async function fetchPendingSwapInfo(
     } else {
       swapType = SWAP_TYPES.INVALID
     }
-    const synthTokenFrom = getTokenByAddress(
-      pendingSwapInfo.synth,
-      chainId,
-    ) as Token
-    const tokenTo = getTokenByAddress(pendingSwapInfo.tokenTo, chainId) as Token
+    const synthTokenFrom = tokens?.[pendingSwapInfo.synth]
+    const tokenTo = tokens?.[pendingSwapInfo.tokenTo]
```

Comment:
it's okay to return null here if either of these tokens are undefined. Just make sure to `console.error` it. This means that you no longer have to add `undefined` to the `PendingSwap` type